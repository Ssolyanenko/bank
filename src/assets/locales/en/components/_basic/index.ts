import modalBox from './modalBox.json';
import cardBlock from './cardBlock.json';
import stepperForm from './stepperForm.json';

export const translationsBasic = {
  ...modalBox,
  ...cardBlock,
  ...stepperForm,
};
