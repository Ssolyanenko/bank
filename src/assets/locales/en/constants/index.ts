import menu from './menu.json';
import errors from './errors.json';
import transferListContent from './transferListContent.json';
import stepForm from './stepForm.json';
import buttonNames from './buttonNames.json';
import lockCardModalContent from './lockCardModalContent.json';
import user from './user.json';
import input from './input.json';
import activateBlockCardContent from './activateBlockCardContent.json';
import resetFormsModalsContent from './resetFormsModalsContent.json';

export const translationsConstants = {
  ...menu,
  ...errors,
  ...transferListContent,
  ...stepForm,
  ...buttonNames,
  ...lockCardModalContent,
  ...user,
  ...input,
  ...activateBlockCardContent,
  ...resetFormsModalsContent,
};
