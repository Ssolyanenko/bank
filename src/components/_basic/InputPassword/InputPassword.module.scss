.formPassword {
  position: relative;
  display: block;

  .showPassword {
    position: absolute;
    top: 50%;
    transform: translateY(-35%);
    right: 7px;

    &:hover {
      cursor: pointer;
    }
  }
}
