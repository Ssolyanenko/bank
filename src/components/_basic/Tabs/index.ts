export * from './Tabs';
export * from './TabTitle';
