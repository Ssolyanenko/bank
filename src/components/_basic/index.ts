export * from './PrimaryButton';
export * from './Button';
export * from './Input';
export * from './InputPassword';
export * from './ModalWrapper';
export * from './ModalConfirmation';
export * from './PhoneCodeTimer';
export * from './ModalBox';
export * from './CloseButton';
export * from './BackButton';
export * from './StepperForm';
export * from './StepperLabel';
export * from './ChatButton';
export * from './BanTimer';
export * from './CardTemplate';
export * from './SecondaryButton';
export * from './generic';
export * from './SelectStyled';
export * from './TextFieldStyled';
export * from './InputStyled';
export * from './CardTemplate';
export * from './Modal';
export * from './SecureWrapper';
export * from './InputRange';
export * from './DropDown';
export * from './RadioButtonsGroup';
export * from './FormError';
export * from './Checkbox';
export * from './Switcher';
export * from './EditButton';
export * from './CardBlock';
export * from './Tabs';
export * from './ModalSuccess';
export * from './DropDownWithCheckbox';
export * from './SelectFormikTemplate';
export * from './Loader';
export * from './DatePicker';
export * from './CustomRoutes';
