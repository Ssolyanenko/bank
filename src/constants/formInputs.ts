import {
  FormAdditionalPhone,
  FormAddress,
  FormBankDelivery,
  FormWork,
  FormUserData,
  CreditPremiumCardValues,
  CardTemplateBillableValues,
} from 'interfaces/creditCardPremium';

export enum AdorPosition {
  START = 'start',
  END = 'end',
}

export enum WorkStatus {
  FOR_HIRE = 'Work for hire',
  FOR_MYSELF = 'Work for myself',
  UNEMPLOYED = 'Unemployed',
}

export enum AdditionalPhone {
  MY_ADDITIONAL = 'My additional phone number',
  RELATIVE = "My relative's phone number",
  FRIEND = "My friend's phone number",
}

export const ValueToBEPhone: { [field: string]: string } = {
  'My additional phone number': 'MY_ADDITIONAL_PHONE',
  "My relative's phone number": 'MY_RELATIVES_PHONE',
  "My friend's phone number": 'MY_FRIENDS_PHONE',
};

export enum NamesToBEPhone {
  TYPE = 'additionalPhoneOwner',
  NUMBER = 'additionalPhoneNumber',
  NUMBER_OWNER = 'additionalPhoneOwnerName',
}

export const ValueToBEWork: { [field: string]: string } = {
  'Work for hire': 'WORK_FOR_HIRE',
  'Work for myself': 'WORK_FOR_MYSELF',
  Unemployed: 'UNEMPLOYED',
  pensioner: 'PENSIONER',
  disabledPerson: 'DISABLED_PERSON',
  lookingForJob: 'LOOKING_FOR_JOB',
  suppByFamily: 'SUPPORTED_BY_FAMILY',
  others: 'OTHERS',
};

export enum NamesToBEWork {
  TYPE = 'placeOfWorkType',
  NAME = 'placeOfWorkName',
  INFORMALLY = 'isWorkInformally',
  OCCUPATION = 'placeOfWorkOccupation',
  UNEMPLOYED = 'unemployedType',
  UNEMPLOYED_OTHERS = 'causeOfUnemployment',
}

export const ValueToBEAddress: { [field: string]: string } = {
  'East of England': 'EAST_OF_ENGLAND',
  London: 'LONDON',
  Midlands: 'MIDLANDS',
  'North East and Yorkshire': 'NORTH_EAST_AND_YORKSHIRE',
  'North West': 'NORTH_WEST',
  'South East': 'SOUTH_EAST',
};

export enum NamesToBEAddress {
  TYPE = 'deliveryType',
  ADDRESS = 'residenceAddress',
  REGION = 'residenceRegion',
  CITY = 'residenceCity',
}

export const ValueToBeMarital: { [field: string]: string } = {
  Single: 'SINGLE',
  Married: 'MARRIED',
  Widowed: 'WIDOWED',
  Divorced: 'DIVORCED',
};

export const ValueToBECity: { [field: string]: string } = {
  Cambridge: 'CAMBRIDGE',
  Liverpool: 'LIVERPOOL',
  London: 'LONDON',
  Manchester: 'MANCHESTER',
};

export enum InputAddPhoneBlock {
  ADDITIONAL_PHONE_NUMBER = 'additionalPhoneNumber',
  ADDITIONAL_PHONE_OWNER = 'additionalPhoneOwner',
  ADDITIONAL_PHONE_OWNER_NAME = 'additionalPhoneOwnerName',
}

export enum InputAddressBlock {
  ADDRESS = 'address',
  POST_CODE = 'postCode',
  CITY = 'city',
  STREET = 'street',
  HOUSE = 'house',
  UNIT = 'unit',
  APARTMENT = 'apartment',
}

export enum InputDeliveryBankBlock {
  BANK_COUNTRY = 'bankCountry',
  BANK_CITY = 'bankCity',
  BANK_BRANCH = 'bankBranchId',
}

export enum InputUserDataBlock {
  FIRST_NAME = 'firstName',
  LAST_NAME = 'lastName',
  EMAIL = 'email',
  PHONE = 'phone',
  PASSPORT = 'passport',
}

export enum InputWorkBlock {
  WORK = 'work',
  WORK_FOR_HIRE_COMPANY = 'workForHireCompany',
  WORK_FOR_HIRE_JOB_TITLE = 'workForHireJob',
  WORK_FOR_MYSELF_CHECK = 'workForMyselfCheck',
  WORK_FOR_MYSELF_OCCUPATION = 'workForMyselfOccupation',
  WORK_FOR_MYSELF_COMPANY = 'workForMyselfCompany',
  UNEMPLOYED = 'unemployed',
  UNEMPLOYED_OTHERS = 'unemployedOthers',
}

export enum InputCreditPremiumField {
  AMOUNT = 'amount',
  INCOME = 'incomePerMonth',
  MARITAL = 'marital',

  BANK_COUNTRY = 'bankCountry',
  BANK_CITY = 'bankCity',
  BANK_BRANCH = 'bankBranchId',
  TERMS_CHECK = 'isTermsAgreed',
}

export enum InputCreditBillableField {
  CATEGORIES = 'categories',
  AMOUNT = 'amount',
  INCOME = 'incomePerMonth',
  MARITAL = 'marital',

  BANK_COUNTRY = 'bankCountry',
  BANK_CITY = 'bankCity',
  BANK_BRANCH = 'bankBranchId',
  IS_TERMS_AGREED = 'isTermsAgreed',
  IS_ACCURACY_CONFIRMED = 'isAccuracyConfirmed',
}

export enum LabelsAddPhoneBlock {
  ADDITIONAL_PHONE_NUMBER = 'Additional phone number',
  ADDITIONAL_PHONE_OWNER = 'Additional phone number owner',
  ADDITIONAL_PHONE_OWNER_NAME = `Owner's name *`,
}

export enum LabelsAddressBlock {
  ADDRESS = 'Actual residence address',
  POST_CODE = 'Post Code',
  CITY = 'Town/City',
  STREAT = 'Street Name',
  HOUSE = 'House Number',
  UNIT = 'Unit',
  APARTMENT = 'Apartment',
}

export enum LabelsDeliveryBankBlock {
  BANK_COUNTRY = 'Country',
  BANK_CITY = 'Town/City',
  BANK_BRANCH = 'Bank branch',
}

export enum LabelsUserDataBlock {
  FIRST_NAME = 'First Name',
  LAST_NAME = 'Last Name',
  EMAIL = 'Email',
  PHONE = 'Phone',
  PASSPORT = 'ID Number',
}

export enum LabelsWorkBlock {
  WORK = 'Place of work',
  WORK_FOR_HIRE_COMPANY = 'Organisation',
  WORK_FOR_HIRE_JOB_TITLE = 'Job Title',
  WORK_FOR_MYSELF_CHECK = 'Work informally',
  WORK_FOR_MYSELF_OCCUPATION = 'Occupation',
  WORK_FOR_MYSELF_COMPANY = 'Organisation',
  UNEMPLOYED = 'Unemployed',
  UNEMPLOYED_OTHERS = 'Cause of unemployment',
}

export enum LabelsCreditPremium {
  AMOUNT = 'Amount',
  INCOME = 'Monthly income',
  MARITAL = 'Marital status',
  TERMS_CHECK = 'The terms of the',
}

export enum LabelsCreditBillable {
  AMOUNT = 'Amount',
  INCOME = 'Monthly income',
  MARITAL = 'Marital status',
  TERMS_CHECK = 'The terms of the',
  CATEGORIES = 'Categories',
}

export enum LabelSecurityForm {
  CURRENT_PASSWORD = 'Enter current password',
  NEW_PASSWORD = 'Create new password',
  CONFIRM_PASSWORD = 'Confirm new password',
  SECURITY_QUESTION = 'Choose security question',
  YOUR_SECURITY_QUESTION = 'Enter your security question',
  ANSWER = 'Enter your security question answer',
}

export enum PlaceholdersWorkBlock {
  WORK = 'Select your place of work ',
  WORK_FOR_MYSELF_OCCUPATION = 'Enter the name of occupation',
  WORK_FOR_HIRE_COMPANY = 'Enter the name of organisation',
  WORK_FOR_HIRE_JOB_TITLE = 'Enter the job title',
  WORK_FOR_MYSELF_COMPANY = 'Enter the name of organisation',
  UNEMPLOYED_OTHERS = 'Enter the cause of unemployment',
}

export enum PlaceholdersAddPhoneBlock {
  ADDITIONAL_PHONE_NUMBER = '+44 1234567899',
  ADDITIONAL_PHONE_OWNER = 'Select an owner of a phone number',
  ADDITIONAL_PHONE_OWNER_NAME = 'Enter the name of an owner',
}

export enum PlaceholdersAddressBlock {
  ADDRESS = 'Select your region',
  POST_CODE = 'Enter your postal code',
  CITY = 'Enter your Town/City',
  STREAT = 'Enter your street name ',
  HOUSE = 'Enter you house number ',
  UNIT = 'Enter you house unit',
  APARTMENT = 'Enter your apartment number',
}

export enum PlaceholdersDeliveryBankBlock {
  BANK_CITY = 'Enter your town/city',
  BANK_BRANCH = 'Select the bank',
}

export enum PlaceholdersCreditPremium {
  AMOUNT = 'Enter the amount',
  INCOME = 'Enter your income',
  MARITAL = 'Select your status ',
  TERMS_CHECK = 'The terms of the ',
}

export enum PlaceholdersCreditBillable {
  AMOUNT = 'Enter the amount',
  INCOME = 'Enter your income',
  MARITAL = 'Select your status ',
  TERMS_CHECK = 'The terms of the ',
  CATEGORIES = 'Select the categories',
}

export enum PlaceholderSecurityForm {
  SECURITY_QUESTION = 'Choose security question',
}

export enum ErrorMessageNames {
  AMOUNT = 'amount',
  POST_CODE = 'post code',
  CITY = 'city',
  STREET = 'street name',
  HOUSE = 'house number',
  UNIT = 'unit',
  APARTMENT = 'flat',
  PHONE = 'phone',
  OWNER = 'owner',
  APARTMENT_NUMBER = 'apartment number',
  BANK_BRANCH = ' ',
}

export const INITIAL_VALUES_USER_DATA_BLOCK: FormUserData = {
  login: '',
  firstName: '',
  lastName: '',
  passport: '',
  phone: '',
  email: '',
  isVip: false,
};

export const INITIAL_VALUES_ADDRESS_BLOCK: FormAddress = {
  address: '',
  postCode: '',
  city: '',
  street: '',
  house: '',
  unit: '',
  apartment: '',
  deliveryType: '',
  residenceCountry: 'GREAT_BRITAN',
  residenceAddress: '',
  residenceRegion: '',
  residenceCity: '',
};

export const INITIAL_VALUES_ADD_PHONE_BLOCK: FormAdditionalPhone = {
  additionalPhoneOwner: '',
  additionalPhoneNumber: '+44 ',
  additionalPhoneOwnerName: '',
};

export const INITIAL_VALUES_WORK_BLOCK: FormWork = {
  work: '',
  workForHireCompany: '',
  workForHireJob: '',
  workForMyselfCheck: false,
  workForMyselfOccupation: '',
  workForMyselfCompany: '',
  unemployed: 'pensioner',
  unemployedOthers: '',
  placeOfWorkType: '',
  placeOfWorkName: '',
  placeOfWorkOccupation: '',
  unemployedType: '',
  causeOfUnemployment: '',
  isWorkInformally: false,
};

export const INITIAL_VALUES_BANK_BLOCK: FormBankDelivery = {
  bankCountry: 'Great Britain',
  bankCity: '',
  bankBranchId: '',
  isTermsAgreed: false,
  isAccuracyConfirmed: false,
};

export const INITIAL_VALUES_CREDIT_PREMIUM: CreditPremiumCardValues = {
  ...INITIAL_VALUES_ADDRESS_BLOCK,
  ...INITIAL_VALUES_ADD_PHONE_BLOCK,
  ...INITIAL_VALUES_WORK_BLOCK,
  ...INITIAL_VALUES_BANK_BLOCK,
  ...INITIAL_VALUES_USER_DATA_BLOCK,
  amount: '',
  incomePerMonth: '',
  marital: '',
};

export const INITIAL_VALUES_CREDIT_BILLABLE: CardTemplateBillableValues = {
  ...INITIAL_VALUES_ADDRESS_BLOCK,
  ...INITIAL_VALUES_ADD_PHONE_BLOCK,
  ...INITIAL_VALUES_WORK_BLOCK,
  ...INITIAL_VALUES_BANK_BLOCK,
  ...INITIAL_VALUES_USER_DATA_BLOCK,
  amount: '',
  categories: [],
  incomePerMonth: '',
  marital: '',
};

export const EnumerationsForm = {
  ADDRESS: [
    { id: 11, text: 'East of England' },
    { id: 12, text: 'London' },
    { id: 13, text: 'Midlands' },
    { id: 14, text: 'North East and Yorkshire' },
    { id: 15, text: 'North West' },
    { id: 16, text: 'South East' },
  ],
  ADDITIONAL_PHONE: [
    { id: 11, text: AdditionalPhone.MY_ADDITIONAL },
    { id: 12, text: AdditionalPhone.RELATIVE },
    { id: 13, text: AdditionalPhone.FRIEND },
  ],
  WORK: [
    { id: 11, text: WorkStatus.FOR_HIRE },
    { id: 12, text: WorkStatus.FOR_MYSELF },
    { id: 13, text: WorkStatus.UNEMPLOYED },
  ],
  MARITAL: [
    { id: 11, text: 'Single' },
    { id: 12, text: 'Married' },
    { id: 13, text: 'Widowed' },
    { id: 14, text: 'Divorced' },
  ],
  BANK_CITY: [
    { id: 11, text: 'Cambridge' },
    { id: 12, text: 'Liverpool' },
    { id: 13, text: 'London' },
    { id: 14, text: 'Manchester' },
  ],
};

export const RadionButtonsWorkBlock = {
  UNEMPLOYED: [
    { id: 11, value: 'pensioner', label: 'Pensioner' },
    { id: 12, value: 'disabledPerson', label: 'Disabled person' },
    { id: 13, value: 'lookingForJob', label: 'Looking for a job' },
    { id: 14, value: 'suppByFamily', label: 'Supported by family' },
    { id: 15, value: 'others', label: 'Others' },
  ],
};

export const ENUMERATIONS_SECURITY_FORM = [
  { id: 1, text: "Mother's maiden name" },
  { id: 2, text: 'Name of childhood best friend' },
  { id: 3, text: 'Favorite book' },
  { id: 4, text: 'Favorite color' },
  { id: 5, text: 'Favorite food' },
  { id: 6, text: 'Add your own question' },
];

export enum SecurityQuestion {
  YOUR_QUESTION = 'Add your own question',
}

export enum DebitCardPlaceholders {
  DELIVERY = 'Delivery',
}

export const ENUMERATION_CATEGORIES = [
  { id: 1, text: 'Online shopping' },
  { id: 2, text: 'Traveling' },
  { id: 3, text: 'Home improvement/furnishings' },
  { id: 4, text: 'Grocery stores' },
  { id: 5, text: 'Wholesales clubs' },
];

export const CARD_DEFAULT_LIMIT = 5000;
export const CARD_LIMIT_AMOUNT = 'limit';
export const CARD_LIMIT_INFO_CHECKBOX = 'limitInfoCheckbox';

export const INITIAL_VALUES_CARD_LIMIT_FORM = {
  [CARD_LIMIT_AMOUNT]: '5,000.00',
  [CARD_LIMIT_INFO_CHECKBOX]: false,
};
