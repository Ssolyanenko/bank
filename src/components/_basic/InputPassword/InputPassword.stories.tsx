import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';

import { InputPassword } from './InputPassword';

export default {
  title: 'Basic/Inputs/InputPassword',
  component: InputPassword,
  argTypes: {
    name: {
      type: 'string',
      defaultValue: 'name',
      description: 'Name of the password input field',
    },
    placeholder: {
      type: 'string',
      defaultValue: 'placeholder',
      description: 'Name of the password input field',
    },
    value: {
      type: 'string',
      defaultValue: 'value',
      description: 'Name of the password input field',
    },
    labelText: {
      type: 'string',
      defaultValue: 'label text',
      description: 'Name of the password input field',
    },
    isError: {
      type: 'boolean',
      defaultValue: false,
      description: 'Name of the password input field',
    },
    onChange: {
      action: 'change',
    },
    onBlur: {
      action: 'blur',
    },
  },
  parameters: {
    docs: {
      description: {
        component: 'Basic password input',
      },
    },
  },
} as ComponentMeta<typeof InputPassword>;

const Template: ComponentStory<typeof InputPassword> = (props) => <InputPassword {...props} />;

export const Default = Template.bind({});
