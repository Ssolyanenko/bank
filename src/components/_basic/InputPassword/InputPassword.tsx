import { useState, FC, ChangeEvent, ReactElement } from 'react';
import { TextField, InputAdornment } from '@mui/material';

import { STATUS_ICONS } from 'constants/statusIcons';
import { AdorPosition } from 'constants/formInputs';
import { PasswordIconInvisible, PasswordIconVisible } from 'assets/loginForm';
import colors from 'styles/variables.module.scss';
import classes from './InputPassword.module.scss';

interface Props {
  labelText?: string;
  name: string;
  value: string;
  isError: boolean;
  placeholder?: string;
  onChange(event: ChangeEvent<HTMLInputElement>): void;
  onBlur?(event: ChangeEvent): void;
  autoComplete?: string;
  InputLabelProps?: object;
}

export const InputPassword: FC<Props> = ({
  labelText = '',
  name,
  onChange,
  onBlur,
  value,
  isError,
  placeholder = '',
  autoComplete = '',
  InputLabelProps = {},
}): ReactElement => {
  const [isShowPassword, setIsShowPassword] = useState(false);

  const onShowPassword = (): void => {
    setIsShowPassword(!isShowPassword);
  };

  return (
    <div className={classes.formPassword}>
      <TextField
        sx={{
          '& .MuiOutlinedInput-root.Mui-focused': {
            '& > fieldset': {
              borderColor: colors.orange,
            },
          },
          '& label': {
            '&.MuiInputLabel-root': {
              color: colors.grayDark_7,
            },
          },
          '& .MuiOutlinedInput-root.Mui-error': {
            paddingRight: '30px',
            '& > fieldset': {
              borderColor: colors.grayDark_7,
            },
          },
        }}
        className={classes.passwordInput}
        variant="outlined"
        type={isShowPassword ? 'text' : 'password'}
        label={labelText}
        name={name}
        onChange={onChange}
        onBlur={onBlur}
        value={value}
        error={isError}
        placeholder={placeholder}
        fullWidth
        autoComplete={autoComplete}
        InputLabelProps={InputLabelProps}
        InputProps={{
          endAdornment: isError && <InputAdornment position={AdorPosition.END}>{STATUS_ICONS.error}</InputAdornment>,
        }}
      />
      <div aria-hidden className={classes.showPassword} onClick={onShowPassword} onKeyDown={onShowPassword}>
        {isShowPassword ? <PasswordIconVisible /> : <PasswordIconInvisible />}
      </div>
    </div>
  );
};
