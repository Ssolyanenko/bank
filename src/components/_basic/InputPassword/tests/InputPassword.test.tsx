import 'jsdom-global/register';
import React from 'react';
import { mount, shallow } from 'enzyme';
import { InputAdornment } from '@mui/material';

import { InputPassword } from 'components/_basic/InputPassword';
import { Default } from 'components/_basic/InputPassword/InputPassword.stories';
import { STATUS_ICONS } from 'constants/statusIcons';
import { AdorPosition } from 'constants/formInputs';

describe('<InputPassword />', (): void => {
  test('should make a snapshot of component', (): void => {
    const props = {
      labelText: 'Password',
      name: 'password',
      value: 'password',
      isError: false,
      onBlur: jest.fn(),
      onChange: jest.fn(),
    };
    const wrapper = shallow(<InputPassword {...props} />);
    const component = wrapper.getElement();

    expect(component).toMatchSnapshot();
  });

  test('should render InputPassword from storybook', (): void => {
    const props = {
      labelText: 'Password',
      name: 'password',
      value: 'password',
      isError: false,
      onBlur: jest.fn(),
      onChange: jest.fn(),
    };

    const wrapper = shallow(<Default {...props} />);

    expect(wrapper.find(InputPassword).length).toBe(1);
  });

  test('should show warning icon when validation doesn`t pass', (): void => {
    const props = {
      labelText: 'Password',
      name: 'password',
      value: 'password',
      isError: true,
      onBlur: jest.fn(),
      onChange: jest.fn(),
    };
    const wrapper = shallow<typeof InputPassword>(<InputPassword {...props} />);
    const visibleIcon = wrapper.find('.passwordInput').prop('InputProps');
    const warningIcon = {
      endAdornment: <InputAdornment position={AdorPosition.END}>{STATUS_ICONS.error}</InputAdornment>,
    };

    expect(visibleIcon).toStrictEqual(warningIcon);
  });

  test('should change state on show password', (): void => {
    const props = {
      labelText: 'Password',
      name: 'password',
      value: 'password',
      isError: false,
      onBlur: jest.fn(),
      onChange: jest.fn(),
    };

    const setStateMock = jest.fn();
    const useStateSpy = jest.spyOn(React, 'useState');
    useStateSpy.mockImplementation((initialState: boolean = false) => [initialState, setStateMock]);

    const wrapper = mount(<InputPassword {...props} />);

    wrapper.find('.showPassword').simulate('click');
    expect(setStateMock).toBeCalledWith(true);
  });
});
