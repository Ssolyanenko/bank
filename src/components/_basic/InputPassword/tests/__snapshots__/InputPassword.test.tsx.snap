// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputPassword /> should make a snapshot of component 1`] = `
<div
  className="formPassword"
>
  <ForwardRef(TextField)
    InputLabelProps={{}}
    InputProps={
      {
        "endAdornment": false,
      }
    }
    autoComplete=""
    className="passwordInput"
    error={false}
    fullWidth={true}
    label="Password"
    name="password"
    onBlur={[MockFunction]}
    onChange={[MockFunction]}
    placeholder=""
    sx={
      {
        "& .MuiOutlinedInput-root.Mui-error": {
          "& > fieldset": {
            "borderColor": "grayDark_7",
          },
          "paddingRight": "30px",
        },
        "& .MuiOutlinedInput-root.Mui-focused": {
          "& > fieldset": {
            "borderColor": "orange",
          },
        },
        "& label": {
          "&.MuiInputLabel-root": {
            "color": "grayDark_7",
          },
        },
      }
    }
    type="password"
    value="password"
    variant="outlined"
  />
  <div
    aria-hidden={true}
    className="showPassword"
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    <PasswordIconInvisible />
  </div>
</div>
`;
